### ISSUE TITLE
Cannot pull public images using docker client without login

### ISSUE THREAD

**Issue by ywk253100** [goharbor/harbor#11266]
```
yinw-a01:cnab-to-oci yinw$ docker logout harbor.10.160.195.133.xip.io
Removing login credentials for harbor.10.160.195.133.xip.io
WARNING: could not erase credentials: error erasing credentials - err: exit status 1, out: `error erasing credentials - err: exit status 1, out: `The specified item could not be found in the keychain.``
yinw-a01:cnab-to-oci yinw$ docker pull harbor.10.160.195.133.xip.io/library/hello-world:latest
Error response from daemon: Get https://harbor.10.160.195.133.xip.io/v2/library/hello-world/manifests/latest: no basic auth credentials
```

Seems that when docker client trying to pull the image, it sends a `/v2/` request first, Harbor responses with status code 401 and header `www-authenticate: Basic realm="harbor"`.
Docker client finds no basic auth credential in local host, it reports an error directly.

**Comment by reasonerjt** [comment]
I'll double check docker's code.

Worst case if it always call `GET /v2` to decide whether to attach credentials to the subsequent requests, we'll have to return the token realm to the client, and decode the token in the middleware.

**Comment by reasonerjt** [comment]
Fixed in #11406 
Verified in build ` v2.0.0-42801b76`
